### ISSUE TITLE
PatchURL for full path - not only query params

### ISSUE THREAD

**Issue by LinuxDoku** [jfyne/live#50]
Nice library so far! Good work, I really enjoy using this on my new application. Thanks for the work.

Is there a reason, why "PatchURL" and its counterpart "HandleParams" only support query parameters and not the full path?

In my use-case I would like to handle the overall navigation with live, so the user clicks an `a`-Tag (e.g. user is on `/list` and clicks `/detail/lorem-ipsum`), which triggers popstate and the server pushes the changed content without a full page reload and the user stays connected with the websocket.

It makes no sense to handle different complex pages with this single handler, but for mostly similar pages with the same events allover this could save some bandwith for the user.

**Comment by jfyne** [comment]
Hi there, sorry I've had a bunch of real life commitments.

There is no reason that they are particularly like that, other than they are trying to replicate Phoenix Liveview, and I can see your use case.

Could you not just use the same live handler at two different paths? Generally I think then you could do most of your database work then in the mount function?

**Comment by LinuxDoku** [comment]
Thanks for you reply. I tested some ways and ended up with a `RouterComponent` which is registered as `/` live handler at the ServeMux. In this component at mount I go through my map of sub-components, each having a route-(pattern) and render the component matching the context's url path.

With the hooks system I added an event handler which intercepts page navigation events and sends a `liverouter--navigate` event with target url back to my router which handles the state change. At the same time I use pushstate to update the url for the user.

So far this works great as it enables navigating around without javascript. Also when the server restarts and/or the socket reconnects the user stays on the same page as the entire state is provided by the socket-url and the server just rerenders and pushes updates to changed data.

I had only some pain with my global http handler, cause all 404 not found requests for missing assets like images, source-maps, etc. triggered live to render the not found component and broke the page state for the current session. It took me some time to figure this out.

When I have the time I will put together an example for the example repo. By the way these examples really helped me to get startet with live - mainly the components are a bit tricky to get started with.
